File: app/src/main/java/com/namazed/beatboxbignerdranch/adapter/SoundAdapter.java
package com.namazed.beatboxbignerdranch.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.recyclerview.widget.RecyclerView;
import com.namazed.beatboxbignerdranch.BeatBox;
import com.namazed.beatboxbignerdranch.R;
import com.namazed.beatboxbignerdranch.Sound;
import org.jetbrains.annotations.NotNull;

import java.util.List;


public class SoundAdapter extends RecyclerView.Adapter<SoundHolder> {

    private final BeatBox beatBox;
    private List<Sound> sounds;

    public SoundAdapter(List<Sound> sounds, BeatBox beatBox) {
        this.sounds = sounds;
        this.beatBox = beatBox;
    }

    @NotNull
    @Override
    public SoundHolder onCreateViewHolder(@NotNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.list_item_sound, parent, false);

        return new SoundHolder(view, beatBox);
    }

    @Override
    public void onBindViewHolder(@NotNull SoundHolder holder, int position) {
        Sound sound = sounds.get(position);
        holder.bindSound(sound);
    }

    @Override
    public int getItemCount() {
        return sounds.size();
    }

}

File: app/src/main/java/com/namazed/beatboxbignerdranch/adapter/SoundAdapter.kt
@file: JvmName("SoundAdapter")
package com.namazed.beatboxbignerdranch.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.namazed.beatboxbignerdranch.BeatBox
import com.namazed.beatboxbignerdranch.R
import com.namazed.beatboxbignerdranch.Sound


class SoundAdapter(private val sounds: List<Sound>, private val beatBox: BeatBox) :
    RecyclerView.Adapter<SoundHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SoundHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(LAYOUT, parent, false)

        return SoundHolder(view, beatBox)
    }

    override fun onBindViewHolder(holder: SoundHolder, position: Int) {
        val sound = sounds[position]
        holder.bindSound(sound)
    }

    override fun getItemCount(): Int {
        return sounds.size
    }


    companion object {

        private val LAYOUT = R.layout.list_item_sound
    }
}